pilot/pkg/config/clusterregistry/secretcontroller.go

```diff
@@ -199,12 +226,32 @@ func addMemberCluster(s *corev1.Secret, cs *ClusterStore) {
 				Namespace: s.ObjectMeta.Namespace,
 			},
 		}
-		cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
-		cs.clusters = append(cs.clusters, &cluster)
+		c.cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
+		c.cs.clusters = append(c.cs.clusters, &cluster)
+		client, _ := kube.CreateInterfaceFromClusterConfig(clientConfig)
+		kubectl := kube.NewController(client, kube.ControllerOptions{
+			WatchedNamespace: c.watchedNamespace,
+			ResyncPeriod:     c.resyncInterval,
+			DomainSuffix:     c.domainSufix,
+		})
+
+		c.serviceController.AddRegistry(
+			aggregate.Registry{
+				Name:             serviceregistry.KubernetesRegistry,
+				ClusterID:        GetClusterID(&cluster),
+				ServiceDiscovery: kubectl,
+				ServiceAccounts:  kubectl,
+				Controller:       kubectl,
+			})
+		stopCh := make(chan struct{})
```

Comment:
the use of this stop channel is problematic. This is another area to look at when the delete is implemented.
the stopCh for the secret controller is also problematic if I'm not mistaken.

Reply:
I will look into it in details while tackling `delete` logic.

Reply:
Please create an issue to track it. thanks.

Reply:
Will do